pkgs/applications/networking/aether/default.nix

```diff
@@ -0,0 +1,114 @@
+{ autoPatchelfHook, makeDesktopItem, lib, stdenv, wrapGAppsHook
+, alsa-lib, at-spi2-atk, at-spi2-core, atk, cairo, cups, dbus, expat, fontconfig
+, freetype, gdk-pixbuf, glib, gtk3, libcxx, libdrm, libnotify, libpulseaudio, libuuid
+, libX11, libXScrnSaver, libXcomposite, libXcursor, libXdamage, libXext
+, libXfixes, libXi, libXrandr, libXrender, libXtst, libxcb, libxshmfence
+, mesa, nspr, nss, pango, systemd, libappindicator-gtk3, libdbusmenu
+, fetchurl, fetchgit, imagemagick
+}:
+stdenv.mkDerivation rec {
+  pname = "aether";
+  version = "2.0.0-dev.15";
+
+  srcs = [
+    (fetchurl {
+      url = "https://static.getaether.net/Releases/Aether-${version}/2011262249.19338c93/linux/Aether-${version}%2B2011262249.19338c93.tar.gz";
+      sha256 = "1hi8w83zal3ciyzg2m62shkbyh6hj7gwsidg3dn88mhfy68himf7";
+      name = "aether-tarball.tar.gz";
+    })
+    (fetchgit {
+      url = git://github.com/aethereans/aether-app;
+      rev = "53b6c8b2a9253cbf056ea3ebb077e0e08cbc5b1d";
+      sha256 = "1kgkzh7ih2q9dsckdkinh5dbzvr7gdykf8yz6h8pyhvzyjhk1v0r";
+    })
+  ];
+
+  sourceRoot = "Aether-${version}+2011262249.19338c93";
+
+  # there is no logo in the tarball so we grab it from github and convert it in the build phase
+  postUnpack =
+    let
+      rawLogo = "aether-app-53b6c8b/aether-core/aether/client/src/app/ext_dep/images/Linux-Windows-App-Icon.png";
+    in
+    ''
+      cp ${rawLogo} ${sourceRoot}
+    '';
+
+  buildPhase =
+    ''
+      convert Linux-Windows-App-Icon.png -resize 512x512 aether.png
+    '';
+
+  dontWrapGApps = true;
+
+  buildInputs = [
+    alsa-lib
+    cups
+    libdrm
+    libuuid
+    libXdamage
+    libX11
+    libXScrnSaver
+    libXtst
+    libxcb
+    libxshmfence
+    mesa
+    nss
+  ];
+
+  nativeBuildInputs = [
+    imagemagick
+    autoPatchelfHook
+    wrapGAppsHook
+  ];
+
+  installPhase =
+  let
+    binaryName = "AetherP2P";
+    desktopItem = makeDesktopItem {
+      name = pname;
+      exec = binaryName;
+      icon = pname;
+      desktopName = "Aether";
+      genericName = meta.description;
+      categories = "Network;";
+      mimeType = "x-scheme-handler/aether";
+    };
```

Comment:
Use copyDesktopItems instead.

Reply:
Can you give a reason why I should do that?

Unfortunately I have serious trouble to find the sources for the imported functions, so I don't know more about `copyDesktopItems` other than that it does not work as a substitute because some argument is missing...

Also please note that the original tarball does not come with a desktop file, so we _have_ to create one. As said I don't know what `copyDesktopItem` does but I assume is copies existing files?

Reply:
Thanks for the explanation.

In contrast to the example, I had to add `runHook postInstall` to the `installPhase` block, otherwise `copyDesktopItems` would just not execute.

Again I have no idea why that is...